aesara/tensor/shape.py

```diff
@@ -885,3 +891,142 @@ def shape_padaxis(t, axis):
     """,
     version=3,
 )
+
+
+def specify_broadcastable(x, *axes):
+    """
+    Specify the input as being broadcastable in the specified axes.
+
+    For example, specify_broadcastable(x, 0) will make the first dimension of
+    x broadcastable. When performing the function, if the length of
+    x along that dimension is not 1, a ValueError will be raised.
+
+    Parameters
+    ----------
+    x : tensor_like
+        Input aesara tensor.
+    axis : an int or an iterable object such as list or tuple of int values
+        The dimension along which the tensor x should be broadcastable.
+        If the length of x along these dimensions is not 1, a ValueError will
+        be raised.
+
+    Returns
+    -------
+    tensor
+        A aesara tensor, which is broadcastable along the specified dimensions.
+
+    """
+    x = as_tensor_variable(x)
+
+    if not axes:
+        return x
+
+    if max(axes) >= x.type.ndim:
+        raise ValueError("Trying to specify broadcastable of non-existent dimension")
+
+    shape_info = [1 if i in axes else None for i in range(len(x.type.shape))]
+    return specify_shape(x, shape_info)
+
+
+class MaskBroadcastable(COp):
```

Comment:
I updated `make_node` to explicitly only mask static dimensions of size 1. This was the original behavior, and it doesn't seem that there is any need in the library to hide other static dimension sizes. Hopefully this `Op` is not here to stay so that's why I tried to make it as limited as possible. The name reflects it.